devtools/JenkinsOnDocker/README.md

```diff
@@ -0,0 +1,165 @@
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+
+# Local Jenkins Setup
+
+Setting up a local Jenkins instance is useful for testing Jenkins job changes
+without affecting the production instance. Using Docker makes the setup very
+straightforward.
+
+## Jenkins on Docker
+
+### Requirements
+If you haven't yet, [install Docker on your machine.](https://docs.docker.com/install/)
+
+
+### Setup using provided scripts
+
+**WARNING: Not for production use.**
+
+*INFO: Changing admin credentials is advised.*
+
+You can utilize scripts in this folder to build Docker container with Jenkins,
+pre-installed plugins and some basic configuration.
+
+```bash
+docker build -t beamjenkins .
+docker run -p 127.0.0.1:8080:8080 beamjenkins:latest
+```
+* plugins.txt -- Pre-assembled list of plugins that are available on Apache
+  Jenkins. This list does not contain versions and might not have all the
+  plugins available. If you find some plugin missing, or having and incorrect
+  version, you can check
+  [infra wiki list](https://cwiki.apache.org/confluence/display/INFRA/Jenkins+Plugin+Upgrades)
+  for plugin updates history. Feel free to update this list with added/removed
+  plugins.
+* docker build -- builds image with name beamjenkins based on Dockerfile located
+  in current folder.
+* docker run -- creates and starts new container.
+    * The `-p 127.0.0.1:8080:8080` parameter to `docker run` ensures that your
+      Jenkins instance will only be available via localhost and not your machine
+      hostname.
+* Default credentials: admin:jenadmin (Suggested to change these in Dockerfile.)
+
+Image built via this scripts will contain all required plugins, basic
+configuration listed in manual setup instructions below and also remap
+JENKINS_HOME, so it does not map to a volume, but is part of container instead.
+
+This last action will allow you to persist changes you do to your container:
+
+```bash
+docker ps -a
+docker commit <container_id> <new_image_name>
+docker run -p 127.0.0.1:8080:8080 <new_image_name>
+```
+* docker ps -- list containers.
+* docker commit -- creates new image based on container.
+
+Such approach may be handy for debugging, but it is highly advised to keep all
+data in docker volumes.
+
+### Manual setup
+
+When invoking Docker, you can map a local directory to persist state and keep
+your Jenkins configuration even after machine reboot. //TODO This is planned to
+change to explaining how to spin up data container.
+
+```bash
+JENKINS_HOME=$(readlink --canonicalize-missing ~/jenkins/homedir)
```

Comment:
I believe that current instructions are generic enough and we still lack solution for Windows. I don't have those OSs at hand to test solutions. 
I suggest to leave this line as-is since it is rather easy to understand and let corresponding platforms owners propose relevant solution.

Reply:
Ok, sounds good to me. We can always extend this later (as you said).